src/libraries/System.Collections/tests/Generic/Queue/Queue.Generic.Tests.cs

```diff
@@ -289,6 +289,59 @@ public void Queue_Generic_TryDequeue_AllElements(int count)
             }
         }
 
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsDrawingSupported), nameof(PlatformDetection.IsNotArm64Process))]
+        public void Queue_Generic_IterateLastIndexOfMaxSizedQueue_DoesNotOverflow()
+        {
+            if (PlatformDetection.IsLinux && PlatformDetection.IsArm64Process)
+            {
+                // on linux-arm64 in CI, the process intermittently gets killed
+                return;
+            }
+
+            Queue<int> queue;
+            try
+            {
+                queue = new Queue<int>(Array.MaxLength);
```

Comment:
Can this be `Queue<byte>` to use much less memory?

Reply:
Yes, we can use byte. Test passes locally on linux-arm64 but gets sigkill on the CI machine. If it fails with byte, I'll just skip it on linux-arm64.